modal/functions.py

```diff
@@ -817,44 +832,44 @@ async def _load(provider: _Function, resolver: Resolver, existing_object_id: Opt
 
             # Create function remotely
             function_definition = api_pb2.Function(
-                module_name=info.module_name,
+                module_name=info.module_name or "",
                 function_name=info.function_name,
                 mount_ids=[mount.object_id for mount in all_mounts],
                 secret_ids=[secret.object_id for secret in secrets],
-                image_id=(image.object_id if image else None),
+                image_id=(image.object_id if image else ""),
                 definition_type=info.definition_type,
-                function_serialized=function_serialized,
-                class_serialized=class_serialized,
+                function_serialized=function_serialized or b"",
+                class_serialized=class_serialized or b"",
                 function_type=function_type,
-                resources=api_pb2.Resources(milli_cpu=milli_cpu, gpu_config=gpu_config, memory_mb=memory),
+                resources=api_pb2.Resources(milli_cpu=milli_cpu, gpu_config=gpu_config, memory_mb=memory or 0),
                 webhook_config=webhook_config,
                 shared_volume_mounts=network_file_system_mount_protos(
                     validated_network_file_systems, allow_cross_region_volumes
                 ),
                 volume_mounts=volume_mounts,
                 proxy_id=(proxy.object_id if proxy else None),
                 retry_policy=retry_policy,
-                timeout_secs=timeout_secs,
-                task_idle_timeout_secs=container_idle_timeout,
-                concurrency_limit=concurrency_limit,
+                timeout_secs=timeout_secs or 0,
+                task_idle_timeout_secs=container_idle_timeout or 0,
+                concurrency_limit=concurrency_limit or 0,
```

Comment:
IIRC there was some discussion recently that container timeouts _can't_ be set to 0. I don't know if that's really a problem here but seems kinda weird.

Reply:
These defaults additions were previously implicit due to protobuf's serialization format, since nullity (?) isn't really a thing there unless you box your primitive types in a `message` type or similar. So for example `task_idle_timeout_secs` is currently determined to be unset by the backend when it's `0`, which isn't great since it prevents us from setting an actual 0-value. According to the docs we have a minimum value of `2` seconds, which is arguably even weirder :D